src/main/java/uk/ac/sanger/sccp/stan/service/PermServiceImp.java

```diff
@@ -27,19 +26,21 @@
     private final OperationTypeRepo opTypeRepo;
     private final OperationRepo opRepo;
     private final MeasurementRepo measurementRepo;
+    private final SlotRepo slotRepo;
 
     @Autowired
     public PermServiceImp(LabwareValidatorFactory labwareValidatorFactory, OperationService opService,
                           WorkService workService,
                           LabwareRepo labwareRepo, OperationTypeRepo opTypeRepo, OperationRepo opRepo,
-                          MeasurementRepo measurementRepo) {
+                          MeasurementRepo measurementRepo, SlotRepo slotRepo) {
```

Comment:
Would you please explain SlotRepo parameter and why is it required as part of this story?

Reply:
The SlotRepo is a data access object to update slots in the database. The PermService now adds the sample from the control labware into a slot of the slide, so it needs to use the slot repo to update the contents of the slot.